### DISCUSSION TITLE
version 9.9.1 and iisnode / web.config

### DISCUSSION THREAD

**Discussion by reggiedunloper** [directus/directus#12998] [Feature Requests]
A heads up to anyone running directus on windows server / via iisnode.

When upgrading directus to version 9.9.1, I ran into an issue where the system would work via localhost (on the server), but not from the actual domain in IIS.  I'm running directus on windows server 2019 and iisnode and been working fine since I initially setup on version 9.1.

IISNode (by default?) uses piped streams instead of ports.  I had my dotenv file setup to use the directus default designated port 8055. My mistake was not realizing that iisnode would then override this with a pipe (as the config occured after the dotenv file). - instead of "localhost:8055", it was actually something like "localhost\\:\pipe\XXXXXX-XXXX-XXXX-XXXX-XXXXXXX"

With the recent fix #12900, the order that config/environment variables are prioritized has updated. IISNode's config would run first, setting the pipe, but then my dotenv file would run directus on port 8055.  Since iisnode wasn't set to use that port, the system would run fine on localhost, but not via my domain in IIS.  

To correct my issue, I removed the port setting from my dotenv file (alternatively, you could probably setup the desired port in the iisnode configuration).

Just wanted to share in case anyone else experiences this issue.

**Comment by AlexVanK** [comment]
This pointed me to the right direction.
I noticed that I started having some "port already in use" messages.

In my case, it helped just killing the node.js process and restart Directus.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by veerendra-shakya** [reply]
how to run directus 10 on iis server with iisnode  

This was working on directus 9 but not working with directus 10
// index.js
var { startServer } = require('directus/server');
startServer();



//web.config
<configuration>
  <system.webServer>
    <handlers>
      <add name="iisnode" path="index.js" verb="*" modules="iisnode" />
    </handlers>
    <rewrite>
      <rules>
        <rule name="iisnode">
          <match url="(.*)" />
          <conditions>
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="True" />
          </conditions>
          <action type="Rewrite" url="index.js" />
        </rule>
      </rules>
    </rewrite>
    <iisnode node_env="production" />
  </system.webServer>
</configuration>
